common/moduleconfig/moduleconfig.go

```diff
@@ -37,6 +40,8 @@ type ModuleConfig struct {
 	Deploy []string `toml:"deploy"`
 	// DeployDir is the directory to deploy from, relative to the module directory.
 	DeployDir string `toml:"deploy-dir"`
+	// GeneratedSchemaDir is the directory to generate protobuf schema files into. These can be picked up by language specific build tools
+	GeneratedSchemaDir string `toml:"generated-schema-dir"`
```

Comment:
what is the function of `GeneratedSchemaDir`? is it needed rather than using the existing `Schema` field in this config?

Reply:
Basically I dump the protos for the external modules I need to generate code for into this dir, and then run the code generation in Maven. It means you can check them into source control and have a build that can run without needing an FTL instance with those modules installed present.

Reply:
could we use the global build directory at `<projectRoot>/.ftl` for this? in Go we put all the external module stubs in `.ftl/go`

Reply:
I could try that later, but this approach follows the standard maven code generation convention. It also allows for different modules in a multi module project to only generate clients for a subset of the modules on the server (this is not implemented yet). I would like to keep this approach for now to see how it works, and possibly re-visit it later.